src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            List<Machine> listOfBulldozers = new ArrayList<>();
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            listOfBulldozers.addAll((bulldozerProducer.get()));
+            return listOfBulldozers;
```

Comment:
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
```
you can create global variable 
`private final MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();`